.github/actions/verify-metrics-snapshot/action.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2023 The Jaeger Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'Verify Metric Snapshot and Upload Metrics'
+description: 'Upload or cache the metrics data after verification'
+inputs:
+  snapshot:
+    description: 'Path to the metric file'
+    required: true
+  artifact_key:
+    description: 'Artifact key used for uploading and fetching artifacts'
+    required: true
+  cache_key:
+    description: 'Cache key used for uploading and fetching the correct cached metric'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Upload Metrics Artifact
+      uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b # v4.5.0
+      with:
+        name: ${{ inputs.artifact_key }}
+        path: ./.metrics/${{ inputs.snapshot }}
+        retention-days: 7
+
+    - name: Cache scraped metrics for tagged release for longer retention
+      if: contains(github.ref,'tags')
+      uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57
+      with:
+        path: ./.metrics/${{ inputs.snapshot }}
+        key: ${{ github.ref_name }}_${{ inputs.artifact_key }}
+
+    - name: Get v2 tag 
+      shell: bash
+      run: |
+        v2_tag=$(make echo-v2)
+        echo "v2_tag=$v2_tag" >> $GITHUB_ENV
+        echo "$v2_tag"
+      
+    - name: Download the cached tagged metrics
+      uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57
+      with:
+        path: ./.metrics/cached_${{ inputs.snapshot}}
+        key: ${{ env.v2_tag }}_${{ inputs.artifact_key }}
+
+    - name: Calculate diff between the snapshots 
+      id: diff-check
+      if: steps.cache-metrics.outputs.cache-hit == 'true'
```

Comment:
where is `cache-metrics` defined? And from the name it sounds like the upload step, but here you want cache download step.